src/Particle.cc

```diff
@@ -459,12 +477,63 @@ Taus::Taus(TTree* _BOOM, std::string filename, std::vector<std::string> syst_nam
   tmp= pstats["Tau2"].dmap.at("DiscrAgainstMuon");
   tau2mu= tmp;
 
-  SetBranch("Tau_idAntiEle", againstElectron);
-  SetBranch("Tau_idAntiMu",  againstMuon);
-  SetBranch("Tau_idDecayMode",  DecayMode);
+  // Get the name of the tau ID algorithm to use.
+  std::string tauIdAlgorithm(pstats["TauID"].smap.at("TauIDAlgorithm"));
+
+  std::cout << "Tau ID algorithm chosen: " << tauIdAlgorithm << std::endl;
+
+  // Check this has either mva or deeptau. Otherwise, go back to MVA for compatibility purposes.
+  try{
+    if(tauIdAlgorithm.find("MVA") == std::string::npos && tauIdAlgorithm.find("DeepTau") == std::string::npos){
+      throw "is unknown or unavailable. Falling back to the MVA-based algorithm (MVAoldDM2017v2).";
+    }
+  }
+  catch(const char* msg){
+
+    std::cout << "ERROR! Tau ID algorithm *** " << tauIdAlgorithm << " *** "  << msg << std::endl; 
+    // --------- Anti-particle discriminators --------- //
+    if(pstats["TauID"].bfind("Is2018")){
```

Comment:
again the year would help?

Reply:
Totally agree. This could definitely be done only once with the year and simplify things.